### ISSUE TITLE
core options

### ISSUE THREAD

**Issue by epayson85** [loganmc10/GLupeN64#47]
Maybe this isn't an issue... and I just don't know how to configure it.

The core specific options.  If I save them per game I get the romname.opt file in the core override folder.  Then it works perfectly for that game.  Is it possible to configure this for the entire core instead of each game?

**Comment by loganmc10** [comment]
I'm not aware of a setting in RetroArch that allows core options per game. The only per-game options I am aware of is controller remaps.

RetroArch creates a file called retroarch-core-options.cfg, that is where it reads the options from. The core itself (GLupeN64) just calls RETRO_ENVIRONMENT_GET_VARIABLE with the key as "glupen64-EnableFBEmulation" for instance, to get what that option is set as. It is up to the frontend to respond to the RETRO_ENVIRONMENT_GET_VARIABLE call with whatever the option it set as.

Basically what I am saying is that GLupeN64 has nothing to do with where the options are stored, it just "speaks" to the frontend using RETRO_ENVIRONMENT_GET_VARIABLE to get the information.

**Comment by loganmc10** [comment]
When I use RetroArch on Android, or Windows, or Linux, or the Raspberry Pi, setting the options via the RetroArch menu (which in turn writes to retroarch-core-options.cfg), sets the options for every game, the are "core-wide". I just need to restart RetroArch for whatever changes I make to take effect.

**Comment by epayson85** [comment]
Well when you go into the core options of Glupen the first option is "create game-options file"  When you do this the specific core options you selected are saved to a file "romname.opt" Then retroarch reads that only when you launch that specific game.  

I can't get these options to save at all unless I do it for each and every game.

**Comment by epayson85** [comment]
So just an fyi again I launch this out of hyperspin.  Hyperspin for some reason causes retroarch to load a different config file.  The config file is located at /data/data/com.retroarch/retroarch.config   I can see the core options cfg there as well but for some reason no matter what I do nothing I change in the options causes the settings to go there.

If I launch retroarch via android and launch an n64 game then it saves to the default core options cfg.  Very odd... even if I edit the hyperspin core options cfg myself it still doesn't read what I put in there.    So this isn't an issue with the core.  It works the correct way.  For something I can't get it to work correctly with hyperspin unless I set it up per game.

**Comment by loganmc10** [comment]
Ya RetroArch doesn't have a way (by itself) to create a per-game core options file, this is some kind of HyperSpin feature.

**Comment by epayson85** [comment]
FIGURED IT OUT! in the override config I added the following -
core_options_path = "/data/data/com.retroarch/retroarch-core-options.cfg"
Then put the core options I wanted in that file.  Works perfectly now!

**Comment by loganmc10** [comment]
OK cool glad you got it figured out
